client/src/components/GetDetails/index.js

```diff
@@ -19,13 +30,49 @@ class GetDetails extends Component {
     });
   };
 
+  toggleOpen = (e) => {
+    const clicked = e.target.value.split('.');
+    if (clicked[1] === 'more') {
+      this.setState({ isOpen: true, selectedCat: clicked[0] });
+    } else {
+      const selected = `selected_${clicked[0]}`;
+      this.setState({ [selected]: e.target.value });
+    }
+  };
+
+  toggleClose = (e) => {
+    e.preventDefault();
+    this.setState({ isOpen: false });
+  };
+
+  changeSelected = (e) => {
+    e.preventDefault();
+    const selected = `selected_${this.state.selectedCat}`;
+    this.setState({ [selected]: e.target.value.split('.')[1], isOpen: false });
+  };
+
   render() {
     const { image_url, apparel, colors } = this.props.location.details;
     return (
       <React.Fragment>
         <Title {...this.props} />
         <Header title="Get your details" />
-        <Form image={image_url} apparel={apparel} colors={colors} />
+        <Form
```

Comment:
Try passing down all state (see below) with {...this.state} and then destructuring them in Form.
```
isOpen={this.state.isOpen}
          selected_itemType={this.state.selected_itemType}
          selected_colors={this.state.selected_colors}
          selected_brands={this.state.selected_brands}
          selected_condition={this.state.selected_condition}
          selected_labelSize={this.state.selected_labelSize}
          selected_age={this.state.selected_age}
          selectedCat={this.state.selectedCat}
          toggleOpen={this.toggleOpen}
          toggleClose={this.toggleClose}
          changeSelected={this.changeSelected}
```